program-runtime/src/timings.rs

```diff
@@ -34,34 +36,59 @@ impl ProgramTiming {
     }
 }
 
-#[derive(Default, Debug)]
+#[derive(Debug)]
+pub enum ExecuteTimingType {
+    CheckUs,
+    LoadUs,
+    ExecuteUs,
+    StoreUs,
+    UpdateStakesCacheUs,
+    NumExecuteBatches,
+    CollectLogsUs,
+}
+
+type Metrics = [u64; 7];
+
+#[derive(Debug, Default)]
 pub struct ExecuteTimings {
-    pub check_us: u64,
-    pub load_us: u64,
-    pub execute_us: u64,
-    pub store_us: u64,
-    pub update_stakes_cache_us: u64,
+    pub metrics: Metrics,
     pub total_batches_len: usize,
-    pub num_execute_batches: u64,
-    pub collect_logs_us: u64,
     pub details: ExecuteDetailsTimings,
     pub execute_accessories: ExecuteAccessoryTimings,
 }
 
 impl ExecuteTimings {
     pub fn accumulate(&mut self, other: &ExecuteTimings) {
-        saturating_add_assign!(self.check_us, other.check_us);
-        saturating_add_assign!(self.load_us, other.load_us);
-        saturating_add_assign!(self.execute_us, other.execute_us);
-        saturating_add_assign!(self.store_us, other.store_us);
-        saturating_add_assign!(self.update_stakes_cache_us, other.update_stakes_cache_us);
-        saturating_add_assign!(self.total_batches_len, other.total_batches_len);
-        saturating_add_assign!(self.num_execute_batches, other.num_execute_batches);
-        saturating_add_assign!(self.collect_logs_us, other.collect_logs_us);
+        if self.metrics.len() == other.metrics.len() {
+            for (i, timing) in self.metrics.iter_mut().enumerate() {
+                saturating_add_assign!(*timing, other.metrics[i]);
+            }
+        }
         self.details.accumulate(&other.details);
         self.execute_accessories
             .accumulate(&other.execute_accessories);
     }
+
+    pub fn saturating_add_in_place(&mut self, timing_type: ExecuteTimingType, value_to_add: u64) {
+        let idx = timing_type as usize;
+        let res = self.metrics[idx].saturating_add(value_to_add);
+        self.metrics[idx] = res;
+    }
+}
+
+impl Index<ExecuteTimingType> for Metrics {
+    type Output = usize;
+    fn index(&self, timing_type: ExecuteTimingType) -> &Self::Output {
+        match timing_type {
+            ExecuteTimingType::CheckUs => &0_usize,
+            ExecuteTimingType::LoadUs => &1_usize,
+            ExecuteTimingType::ExecuteUs => &2_usize,
+            ExecuteTimingType::StoreUs => &3_usize,
+            ExecuteTimingType::UpdateStakesCacheUs => &4_usize,
+            ExecuteTimingType::NumExecuteBatches => &5_usize,
+            ExecuteTimingType::CollectLogsUs => &6_usize,
+        }
+    }
```

Comment:
We can just return a reference into the array directly here.  Probably need `IndexMut` instead though